mvc/views/tag-helpers/intro.rst

```diff
@@ -1,8 +1,275 @@
+
+A pure HTML ``<label>`` tag displays the HTML tag (with the default Visual Studio color theme) in a brown font, the attributes in red, and the attribute values in blue.
+
+.. image:: intro/_static/LableHtmlTag.png 
+
+After you enter ``<label``, IntelliSense lists the available HTML/CSS attributes and the Tag Helper-targeted attributes:
+
+.. image:: intro/_static/labelattr.png
+
+IntelliSense statement completion allows you to enter the tab key to complete the statement with the selected value:
+
+.. image:: intro/_static/stmtcomplete.png
+
+As soon as a Tag Helper attribute is entered, the tag and attribute fonts change. Using the default Visual Studio "Blue" or "Light" color theme, the font is bold purple. If you're using the "Dark" theme the font is bold teal. The images in this document were taken using the default theme.
+
+.. image:: intro/_static/labelaspfor.png
+
+You can enter the Visual Studio *CompleteWord* shortcut (Ctrl +spacebar is the `default <https://msdn.microsoft.com/en-us/library/da5kh0wa.aspx>`__) inside the double quotes (""), and you are now in C#, just like you would be in a C# class. IntelliSense displays all the methods and properties on the page model. The methods and properties are available because the property type is ``ModelExpression``. In the image below, I'm editing the ``Register`` view, so the ``RegisterViewModel`` is available.
+
+.. image:: intro/_static/intellemail.png
+
+IntelliSense lists the properties and methods available to the model on the page. The rich IntelliSense environment helps you select the CSS class:
+
+.. image:: intro/_static/iclass.png
+
+.. image:: intro/_static/intel3.png
+
+Tag Helpers compared to HTML Helpers
+---------------------------------------------
+
+Tag Helpers attach to HTML elements in Razor views, while `HTML Helpers <http://stephenwalther.com/archive/2009/03/03/chapter-6-understanding-html-helpers>`__ are invoked as methods interspersed with HTML in Razor views. Consider the following Razor markup, which creates an HTML label with the CSS class "caption":
+
+.. code-block:: html
+
+	@Html.Label("FirstName", "First Name:", new {@class="caption"})
+
+The at (``@``) symbol tells Razor this is the start of code. The next two parameters ("FirstName" and "First Name:") are strings, so `IntelliSense <https://msdn.microsoft.com/en-us/library/hcw1s69b.aspx>`_ can't help. The last argument:
+
+.. code-block:: html
+
+  new {@class="caption"}
+  
+Is an anonymous object used to represent attributes. Because **class** is a reserved keyword in C#, you use the ``@`` symbol to force C# to interpret "@class=" as a symbol (property name). To a front-end designer (someone familiar with HTML/CSS/JavaScript and other client technologies but not familiar with C# and Razor), most of the line is foreign. The entire line must be authored with no help from IntelliSense.
+  
+Using the ``LabelTagHelper``, the same markup can be written as:
+
+.. image:: intro/_static/label2.png 
```

Comment:
I need a service to:
ren *.PNG *.png
A couple images were .PNG and referenced as *.png
They really need to fix *nix and other OSes. It's beyond absurd to differentiate a file on case.